packages/clangd-contexts/src/clangd-config.ts

```diff
@@ -0,0 +1,102 @@
+/********************************************************************************
+ * Copyright (c) 2021 STMicroelectronics and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ *******************************************************************************/
+
+import * as fs from 'fs';
+import * as yaml from 'js-yaml';
+import { isFile, toConfigPath } from './util/file-util';
+
+/** Description of a project configuration as encoded in a `.clangd` file. */
+export interface ClangdConfig {
+    CompileFlags: CompileFlags;
+    SourceFile: string;
```

Comment:
`SourceFile` has a bit of an odd ring in this context, since for a compiler that could mean the source code. Perhaps `configFile`?